testing/bdd_examples/only_test.ts

```diff
@@ -0,0 +1,221 @@
+/** Please note that this code should not be committed. Used here as demonstration. */
+
+import { assertEquals } from "../asserts.ts";
+import { describe, it } from "../bdd.ts";
+
+const onlyTests = describe.only("Only");
+
+const anonTests = describe("Anon");
+
+const suiteOnlyTests = describe({ name: "Suite", suite: onlyTests });
+
+/**
+ * Normal - this case already works properly without any change
+ * Fixed - this case has been fixed in this PR
+ * Bugged - this case still doesn't run after the fix
+ */
+
+describe(onlyTests, "describe 1.1", () => {
+  /**
+   * Should run
+   * Fixed
+   */
+  it("it 1.1", () => {
+    assertEquals(0, 0);
+  });
+});
+
+/**
+ * Should run
+ * Normal
+ */
+it(onlyTests, "it 4.1", () => {
+  assertEquals(0, 0);
+});
+
+/**
+ * Should run
+ * Fixed
+ */
+it(suiteOnlyTests, "it 4.2", () => {
+  assertEquals(0, 0);
+});
+
+/**
+ * Should not run
+ * Normal
+ */
+it(anonTests, "it 4.3", () => {
+  assertEquals(0, 0);
+});
+
+/**
+ * Should not run
+ * Normal
+ */
+it("it 4.4", () => {
+  assertEquals(0, 0);
+});
+
+describe("describe 1.2", () => {
+  /**
+   * Should run
+   * Normal
+   */
+  it(onlyTests, "it 1.2.1", () => {
+    assertEquals(0, 0);
+  });
+  /**
+   * Should not run
+   * Normal
+   */
+  it(anonTests, "it 1.2.2", () => {
+    assertEquals(0, 0);
+  });
+  /**
+   * Should run
+   * Fixed
+   */
+  it(suiteOnlyTests, "it 1.2.3", () => {
+    assertEquals(0, 0);
+  });
+  /**
+   * Should not run
+   * Normal
+   */
+  it("it 1.2.4", () => {
+    assertEquals(0, 0);
+  });
+});
+
+describe(onlyTests, "outer 2.1", () => {
```

Comment:
@kt3k This one is similar to the one you mentioned, although the suite is linked in the outer describe, rather than the inner describe. I think we can leave this one out if it's not a supported case.